File: kotlin/modules/srf/src/main/kotlin/info/agilite/srf/domain/Srf2060Models.kt
package info.agilite.srf.domain

import info.agilite.boot.orm.annotations.DbSimpleJoin
import info.agilite.boot.orm.annotations.DbTable
import info.agilite.shared.entities.cgs.Cgs15
import info.agilite.shared.entities.cgs.Cgs80
import info.agilite.shared.entities.gdf.Gdf10
import info.agilite.shared.entities.scf.Scf02
import info.agilite.shared.entities.srf.Srf01

@DbTable("srf01")
@DbSimpleJoin("srf01entidade, srf01natureza, INNER:cgs18modeloEmail")
data class Srf2060Mail(
  val cgs18id: Long,
  val cgs18nome: String,

  val cgs15id: Long,
  val cgs15nome: String,
  val cgs15titulo: String,

  val cgs80email: String?,

  val srf01id: Long,
  val srf01nome: String,
){
  fun validarPraEnviarEmail(): String? {
    if(cgs80email == null) return "Email não informado para o documento ${srf01nome}"
    return null
  }
}

class Srf2060Doc(
  val srf01: Srf01,
  val cgs80: Cgs80,
  val gdf10: Gdf10,
  val cgs15: Cgs15,
  val scf02s: MutableList<Scf02>
)

File: flutter/modules/srf/lib/tasks/srf2060/srf2060_models.dart
class Srf2060Mail {
  final int cgs18id;
  final String cgs18nome;
  final int cgs15id;
  final String cgs15nome;
  final String cgs15titulo;
  final String? cgs80email;
  final int srf01id;
  final String srf01nome;
  final int srf01integracaoScf;
  final int srf01integracaoGdf;

  Srf2060Mail({
    required this.cgs18id,
    required this.cgs18nome,
    required this.cgs15id,
    required this.cgs15nome,
    required this.cgs15titulo,
    this.cgs80email,
    required this.srf01id,
    required this.srf01nome,
    required this.srf01integracaoScf,
    required this.srf01integracaoGdf,
  });

  factory Srf2060Mail.fromJson(Map<String, dynamic> json) {
    return Srf2060Mail(
      cgs18id: json['cgs18id'] as int,
      cgs18nome: json['cgs18nome'] as String,
      cgs15id: json['cgs15id'] as int,
      cgs15nome: json['cgs15nome'] as String,
      cgs15titulo: json['cgs15titulo'] as String,
      cgs80email: json['cgs80email'] as String?,
      srf01id: json['srf01id'] as int,
      srf01nome: json['srf01nome'] as String,
      srf01integracaoScf: json['srf01integracaoScf'] as int,
      srf01integracaoGdf: json['srf01integracaoGdf'] as int,
    );
  }
}
